### PR TITLE
Remove CeedOperatorSetNumQuadraturePoints

### PR DESCRIPTION
**Pull request by jeremylt**
This should be merged right before we cut the new release.

### CODE REVIEW

File: interface/ceed-operator.c [hunk 1]
```diff
@@ -757,10 +757,10 @@ int CeedOperatorSetField(CeedOperator op, const char *field_name, CeedElemRestri
     op->has_restriction = true;  // Restriction set, but num_elem may be 0
   }
   CeedCall(CeedBasisReferenceCopy(b, &(*op_field)->basis));
-  if (op->num_qpts == 0) CeedCall(CeedOperatorSetNumQuadraturePoints(op, num_qpts));
-
+  op->num_qpts = num_qpts;
```
Comment (jrwrigh) [inline interface/ceed-operator.c:760]: Maybe we should move this up to line 712, where `num_qpts` is checked against it's previous value? I feel like this makes more sense up there than here. (plus I spent about 15 minutes trying to figure out whether removing the `if (op->num_qpts == 0)` had any consequence on the code. I think this on line 712 would've prevented that, as it puts the setting of the value right by where it gets checked).

**Review by jrwrigh — APPROVED** [review]
Minor comment on code placement

File: interface/ceed-operator.c (hunk 1 above)
Comment (jrwrigh) [reply interface/ceed-operator.c:760]: Maybe even keeping the if statement to show intent would be a good idea. ie. we're not going to reset the value every single call. It should only ever be one value.

Comment (jeremylt) [reply interface/ceed-operator.c]: I put back the `if (op->num_qpts == 0)` since we cannot move this logic

**Pull request merged**